packages/account-v2/src/modules/FinancialAssessment/__tests__/FinancialAssessmentForm.spec.tsx

```diff
@@ -0,0 +1,230 @@
+        (useFinancialAssessment as jest.Mock).mockReturnValue({
+            data: {},
+            error: {
+                error: {
+                    message: 'Error Loading API',
+                },
+            },
+            isLoading: false,
+            mutation: {
+                error: undefined,
+                isLoading: false,
+                isSuccess: false,
+            },
+            update: jest.fn(),
+        });
+        render(<FinancialAssessmentForm />);
+        expect(screen.getByText('Error Loading API')).toBeInTheDocument();
+    });
+
+    it('render Update Error Message on update error', () => {
+        (useFinancialAssessment as jest.Mock).mockReturnValue({
+            data: {},
+            error: undefined,
+            isLoading: false,
+            mutation: {
+                error: {
+                    error: {
+                        message: 'Error Updating API',
+                    },
+                },
+                isLoading: false,
+                isSuccess: false,
+            },
+            update: jest.fn(),
+        });
+        render(<FinancialAssessmentForm />);
+        expect(screen.getByText('Error Updating API')).toBeInTheDocument();
+    });
+
+    it('renders Financial Assessment form', () => {
+        render(<FinancialAssessmentForm />);
+        expect(screen.getByText('Financial information')).toBeInTheDocument();
+        expect(screen.getByText('(All fields are required)')).toBeInTheDocument();
+        expect(screen.getByRole('combobox', { name: 'Source of income' })).toBeInTheDocument();
+        expect(screen.getByRole('combobox', { name: 'Source of wealth' })).toBeInTheDocument();
+    });
+
+    it('calls update function with valid values on form submission', async () => {
+        render(<FinancialAssessmentForm />);
+
+        act(() => {
+            const wealthSourceField = screen.getByRole('combobox', { name: 'Source of wealth' });
+            userEvent.type(wealthSourceField, 'Inheritance');
+        });
+
+        act(() => {
+            const inheritanceOption = screen.getByRole('option', { name: 'Inheritance' });
+            userEvent.click(inheritanceOption);
+        });
+
+        const submitButton = screen.getByRole('button', { name: 'Submit' });
+
+        userEvent.click(submitButton);
+
+        await waitFor(() => {
+            expect(mockUpdate).toHaveBeenCalledWith({
+                ...financialAssessmentPayload,
+                occupation: 'Developer',
+                source_of_wealth: 'Inheritance',
+            });
+        });
+    });
+
+    it('calls update function without occupation if employment status un-employed on form submission', async () => {
+        render(<FinancialAssessmentForm />);
+
+        act(() => {
+            const employmentStatusField = screen.getByRole('combobox', { name: 'Employment status' });
+            userEvent.type(employmentStatusField, 'Unemployed');
```

Comment:
we don't need to wrap user events in act, it is wrapped under the hood. it was not working without act?

Reply:
In some of the cases, react-testing-library was throwing a warning to wrap it with act, I am not sure why, it could be because i am firing multiple actions and every one is dependent on the other.

Reply:
so it is better to wrap expectations in waitFor, have you tried it?

Reply:
Updated.